vm/vm.go

```diff
@@ -116,6 +120,13 @@ func (vm *VM) Run(program *Program, env interface{}) (out interface{}, err error
 		case OpFetchEnvFast:
 			vm.push(env.(map[string]interface{})[program.Constants[arg].(string)])
 
+		case OpFetcher:
+			a := vm.pop()
+			vm.push(a.(Fetcher).Fetch(program.Constants[arg].(string)))
+
+		case OpFetcherEnv:
+			vm.push(env.(Fetcher).Fetch(program.Constants[arg].(string)))
```

Comment:
The resulting program for the real_world bench test uses this opcode. And looks like proper usage of reflect can be faster.